lib/oli_web/components/delivery/actions/actions.ex

```diff
@@ -54,100 +54,156 @@ defmodule OliWeb.Components.Delivery.Actions do
+                      <%= elem.title %>
+                    </option>
+                  <% end %>
+                </select>
+              </form>
+            </div>
+            <%= if @section.requires_payment and @is_admin do %>
+              <div class="flex justify-between items-center">
+                <div class="flex flex-col">
+                  <span class="dark:text-black">Bypass payment</span>
+                  <span class="text-xs text-gray-400 dark:text-gray-950">Apply bypass payment</span>
+                </div>
+                <button
+                  class="torus-button flex justify-center primary h-9 w-48"
+                  disabled={@has_payment}
+                  phx-click="display_bypass_modal"
+                  phx-target={@myself}
+                >
+                  Apply Bypass Payment
+                </button>
+              </div>
+              <div :if={@has_payment} class="flex justify-between items-center">
+                <div class="flex flex-col">
+                  <span class="dark:text-white">Update payment status</span>
+                </div>
+                <form
+                  id={"payment_status_form_#{@payment_status_uuid}"}
+                  phx-change={Modal.show_modal("payment_status_modal")}
+                >
+                  <select class="torus-select pr-32" name="payment_status">
+                    <option selected={true} value={:paid}>
+                      Paid
+                    </option>
+                    <option selected={false} value={:unpaid}>
+                      Unpaid
+                    </option>
+                  </select>
+                </form>
+              </div>
+            <% end %>
+
+            <%= if @is_admin do %>
+              <.live_component
+                id="transfer_enrollment"
+                module={OliWeb.Delivery.Actions.TransferEnrollment}
+                section={@section}
+                user={@user}
+              />
+            <% end %>
+
+            <div class="ml-auto">
+              <button phx-click={Modal.show_modal("unenroll_user_modal")} class="btn btn-danger">
+                Unenroll
               </button>
             </div>
-          <% end %>
-
-          <%= if @is_admin do %>
-            <.live_component
-              id="transfer_enrollment"
-              module={OliWeb.Delivery.Actions.TransferEnrollment}
-              section={@section}
-              user={@user}
-            />
-          <% end %>
-
-          <div class="ml-auto">
-            <button phx-click={Modal.show_modal("unenroll_user_modal")} class="btn btn-danger">
-              Unenroll
-            </button>
           </div>
-        </div>
-      <% end %>
+        <% end %>
+      </div>
     </div>
     """
   end
 
+  def handle_event("force_reset_payment_status", _params, socket) do
+    {:noreply, assign(socket, payment_status_uuid: UUID.uuid4())}
+  end
```

Comment:
Any reason why we are resetting the form ID after canceling the modal?

Reply:
Yes, I wanted to force the browser to set the select input back to its original value in case the admin cancels or closes the confirmation modal. A "hacky" way of addressing this is forcing Phoenix to render the component again (-> changing its id). 
In the attached video you can see this behaviour in action.

Reply:
I got it, otherwise the form is going to stuck with an incorrect value. Nice work.